mipengine/controller/algorithm_executor.py

```diff
@@ -108,27 +113,81 @@ def __init__(
         )
 
     def run(self):
-        algorithm_result = self.algorithm_flow_module.run(self.execution_interface)
+        try:
```

Comment:
I think that these changes are not needed at all.

Errors should be handled only in the quart error_handler, not externally. 

The generic error handler could be improved to print the errors like:
```
@error_handlers.app_errorhandler(Exception)
def handle_unexpected_exception(error: Exception):
    print(
        f"Internal Server Error."
        f"\nErrorType: {type(error)}"
        f"\nError: {error}"
        f"\nTraceback: {traceback.print_tb(error.__traceback__)}"
    )

    return "", HTTPStatusCode.UNEXPECTED_ERROR
```

This approach doesnt print the traceback for some reason, but I guess it's due to the background thread logic.

Reply:
these exceptions are algorithm execution specific, they convey information about the algorithm execution. Τhe webapi(the quart) should only be concerned with triggering the execution, as soon as it gets a request to execute an algorithm and then sending back a result as a reply to the request it received. What would maybe make sense though is instead of forming a fail result here `return error_result`, raise an specific exception, like `AlgorithmTimeoLimitExceeded` which can then be handled by the error_handlers.py

Reply:
The quart error_handler is also responsible for returning the proper status code. There is a default exception handler that resolves to 500. If the exception is caught in the algorithm_executor and a "normal" response is returned, it won't work properly, a 200 will be returned.

Raising a different exception also works, if we don't want to raise an "internal" exception. But again these errors should not be handled here, if we want to use the node task interface properly, the exceptions should be handled internally in the decorator and a "generic_algo_executor" exception be raised internally. That "generic_algo_executor" should be caught by the default quart error handler.

Reply:
Now all 3 exceptions are caught here, in case the algorithm_executor needs to act differently on each one and it then throws a "generic" `AlgorithmExecutionException` which is caght by the webapi exception handler.